src/shared/libfido2-util.c

```diff
@@ -519,6 +638,30 @@ int fido2_use_hmac_hash(
                         goto finish;
                 }
 
+                r = fido2_is_cred_in_specific_token(path, rp_id, cid, cid_size, pins, required);
+                /* We handle PIN related errors here since we have used PIN in the check.
+                 * If PIN error happens, we can avoid pin retries decreased the second time. */
+                if (IN_SET(r, -ENOANO, -ENOLCK, -EOWNERDEAD)) {
+                        /* If it's not the last device, try next one */
+                        if (i < found - 1)
+                                continue;
+                        else {
```

Comment:
you can remove this else indentation, after all the if branch already die `continue`. We are big fans of shallow indentation.